.github/workflows/ci.yaml

```diff
@@ -0,0 +1,112 @@
+            matrix:
+                php-version:
+                    - '7.4'
+
+        steps:
+            -
+                name: "Checkout code"
+                uses: "actions/checkout@v2"
+
+            -
+                name: "Install PHP"
+                uses: "shivammathur/setup-php@v2"
+                with:
+                    coverage: "none"
+                    php-version: "${{ matrix.php-version }}"
+                    tools: "composer:v2"
+
+            -
+                name: "Validate composer.json"
+                run: "composer validate --strict --no-check-lock"
+
+            -
+                name: "Composer install"
+                uses: "ramsey/composer-install@v1"
+                with:
+                    composer-options: "--no-suggest --no-scripts"
+
+    tests:
+        needs:
+            - "composer-validate"
+
+        name: "PHP ${{ matrix.php-version }} + ${{ matrix.dependencies }} + Composer v${{ matrix.composer-version }}"
+
+        runs-on: ubuntu-16.04
+
+        continue-on-error: ${{ matrix.allow-failures }}
+
+        strategy:
+            fail-fast: false
+            matrix:
+                php-version:
+                    - '7.1'
+                    - '7.2'
+                    - '7.3'
+                    - '7.4'
+                    - '8.0'
+                dependencies:
+                    - 'lowest'
+                    - 'highest'
+                composer-version:
+                    - '1'
+                    - '2'
+                allow-failures: [false]
+
+        steps:
+            - name: "Checkout code"
+              uses: actions/checkout@v2.3.3
+
+            - name: "Install PHP with extensions"
+              uses: shivammathur/setup-php@2.7.0
+              with:
+                  coverage: "none"
+                  php-version: ${{ matrix.php-version }}
+                  tools: composer:v${{ matrix.composer-version }}
+
+            - name: "Add PHPUnit matcher"
+              run: echo "::add-matcher::${{ runner.tool_cache }}/phpunit.json"
+
+            - if: matrix.php-version == '8.0'
+              run: composer config platform.php 7.4.99 || true
+
+            - if: matrix.composer-version == '1'
+              run: composer require --no-update composer/composer:^1.0.2
+
+            - if: matrix.composer-version == '2'
+              run: composer require --no-update composer/composer:^2
+
+            - name: "Composer install"
+              uses: "ramsey/composer-install@v1"
+              with:
+                  dependency-versions: "${{ matrix.dependencies }}"
+
+            - name: "Install PHPUnit"
+              run: vendor/bin/simple-phpunit install
+
+            - name: "PHPUnit version"
+              run: vendor/bin/simple-phpunit --version
+
+            - name: "Run tests"
+              run: vendor/bin/simple-phpunit
+
+            - name: "Lint PHP files"
+              run: find src/ -name '*.php' | xargs -n1 php -l
```

Comment:
@nicolas-grekas do we really need to do this for each job of the matrix, do we need this for every php version or is it enough todo it once, like we do for composer validate?

Reply:
why not? we need it for 7.1 for sure, is it an issue to do it on all versions?

Reply:
>  is it an issue to do it on all versions?

No but we are doing it multiple times for one php version because of the matrix

Reply:
nice, that fits with my other comments: let's kill this matrix and make it a list

Reply:
Done in a separate job only once per version